src/test/java/nl/tudelft/ewi/devhub/server/backend/DeliveriesBackendTest.java

```diff
@@ -0,0 +1,365 @@
+		
+		when(deliveriesDAO.getDeliveries(Matchers.eq(assignment), Matchers.eq(group))).thenReturn(deliveries);
+		
+		when(attachment.getPath()).thenReturn(pathName + fileName);
+	}
+	
+	@Test
+	public void deliveryIsStored() throws UnauthorizedException, ApiError {
+		this.deliveriesBackend.deliver(delivery);
+		
+		verify(deliveriesDAO).persist(delivery);
+	}
+	
+	@Test
+	public void adminCanStoreDelivery() throws UnauthorizedException, ApiError {
+		isAnAdmin();
+		
+		this.deliveriesBackend.deliver(delivery);
+		
+		verify(deliveriesDAO).persist(delivery);
+	}
+
+	private void isAnAdmin() {
+		when(currentUser.isAssisting(Matchers.eq(course))).thenReturn(false);
+		when(group.getMembers()).thenReturn(Sets.newHashSet());
+	}
+	
+	@Test
+	public void assistantCanStoreDelivery() throws UnauthorizedException, ApiError {
+		isAnAssistant();
+		
+		this.deliveriesBackend.deliver(delivery);
+		
+		verify(deliveriesDAO).persist(delivery);
+	}
+
+	private void isAnAssistant() {
+		when(currentUser.isAdmin()).thenReturn(false);
+		when(group.getMembers()).thenReturn(Sets.newHashSet());
+	}
+	
+	@Test
+	public void groupMemberCanStoreDelivery() throws UnauthorizedException, ApiError {
+		isAGroupMember();
+		
+		this.deliveriesBackend.deliver(delivery);
+		
+		verify(deliveriesDAO).persist(delivery);
+	}
+
+	private void isAGroupMember() {
+		when(currentUser.isAdmin()).thenReturn(false);
+		when(currentUser.isAssisting(Matchers.eq(course))).thenReturn(false);
+	}
+	
+	@Test(expected=UnauthorizedException.class)
+	public void unauthorizedWhenNoAdminNoAssistantOrNoGroupMember() throws ApiError {
+		hasNoPermission();
+		
+		this.deliveriesBackend.deliver(delivery);
+	}
+
+	private void hasNoPermission() {
+		when(currentUser.isAdmin()).thenReturn(false);
+		when(currentUser.isAssisting(Matchers.eq(course))).thenReturn(false);
+		when(group.getMembers()).thenReturn(Sets.newHashSet());
+	}
+	
+	@Test(expected=IllegalStateException.class)
+	public void whenAlreadyApprovedOrDisapprovedRejectDelivery() throws UnauthorizedException, ApiError {
+		when(deliveriesDAO.lastDeliveryIsApprovedOrDisapproved(Matchers.eq(assignment), Matchers.eq(group)))
+			.thenReturn(true);
+		
+		this.deliveriesBackend.deliver(delivery);
+	}
+	
+	@SuppressWarnings("unchecked")
+	@Test(expected=ApiError.class)
+	public void whenStoringDeliveryFailedErrorToUser() throws UnauthorizedException, ApiError {
+		when(deliveriesDAO.persist(Matchers.eq(delivery))).thenThrow(Exception.class);
+		
+		this.deliveriesBackend.deliver(delivery);
+	}
+	
+	@Test
+	public void attachFileToDelivery() throws UnauthorizedException, ApiError {
+		attachedFile();
+	}
+
+	private void attachedFile() throws ApiError {
+		this.deliveriesBackend.attach(delivery, fileName, in);
+		
+		assertEquals(1, delivery.getAttachments().size());
+		// Because a simple get did not return the correct deliveryAttachment, we had to do it the fancy way.
+		assertEquals(1, delivery.getAttachments().stream()
```

Comment:
assertEquals(fileName, delivery.getAttachments().iterator().next().getFileName()) did not do the trick?